rest_framework/fields.py

```diff
@@ -1042,10 +1042,16 @@ def to_internal_value(self, data):
         except KeyError:
             self.fail('invalid_choice', input=data)
 
+    def representation_value(self, value):
+        try:
+            return self.choice_strings_to_values[six.text_type(value)]
+        except KeyError:
+            return value
```

Comment:
That was actually my first attempt, although how would you inline a try/except inside a list comprehension?  Was asking on #django and was told it may be possible with a generator expression & yielding?